src/dependencies/BuildComponent.js

```diff
@@ -5,7 +5,8 @@ export function makeDisplayName(
   prefix: string,
   BaseComponent: { displayName: string }
```

Comment:
@theopak the [typechecks are failing](https://travis-ci.org/HubSpot/general-store/builds/264796158#L606-L615) on travis ci. Maybe you could change this to something like...

```js
// ...
  BaseComponent: { displayName?: string, name?: string }
) {
  const baseComponentName = BaseComponent.displayName || BaseComponent.name || 'Component';
// ...
```